From c844e8011f0726e2a371115c209d4c3d63273b3b Mon Sep 17 00:00:00 2001
From: Daniel Mihai <dmihai@microsoft.com>
Date: Thu, 1 Sep 2022 15:54:16 +0000
Subject: [PATCH 10/10] Merged PR 9806: Fix enable_debug for [hypervisor.clh]

Fix error when using enable_debug = true in configuration.toml:

level=error msg="Error create pseudo tty"
error="open /dev/ptmx: operation not permitted"
---
 src/runtime/pkg/resourcecontrol/cgroups.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/runtime/pkg/resourcecontrol/cgroups.go b/src/runtime/pkg/resourcecontrol/cgroups.go
index d4608458..f674e97a 100644
--- a/src/runtime/pkg/resourcecontrol/cgroups.go
+++ b/src/runtime/pkg/resourcecontrol/cgroups.go
@@ -57,6 +57,7 @@ func sandboxDevices() []specs.LinuxDeviceCgroup {
 		"/dev/zero",
 		"/dev/urandom",
 		"/dev/console",
+		"/dev/ptmx",
 	}
 
 	// Processes running in a device-cgroup are constrained, they have acccess
-- 
2.17.1

